File: search/二分查找/binary_search.py
def binarySearch(nums,key):
    low = 0
    hight = len(nums)-1
    while low <=hight:
        m = (low + hight) // 2
        if nums[m]<key:
            low = m + 1
        elif nums[m]>key:
            hight = m-1
        else:
            return m
    return

nums = [1,2,3,4,5,6,7,8,9,10]
key = 2
print "Input:",nums
print "Key:", key
print "Result Index:",binarySearch(nums,key)

File: search/二分查找/binary_search.java
public class binary_search {
    public static void main(String []args){
        int[] array = {1,2,3,4,5};
        int key = 2;
        System.out.println("Input:");
        for(int a:array)
            System.out.println(a);
        System.out.println("Key:"+key+"");
        System.out.println("Result Index:"+binarySearch(array,key)+"");
    }
    // binarySearch 二分查找
    public static int binarySearch(int []arr,int key){
        int low = 0;
        int hight = arr.length-1;
        while (low<=hight){
            int m = (low+hight) / 2;
            if (arr[m]<key) {
                low = m + 1;
            }else if (arr[m]>key){
                hight = m-1;
            }else{
                return m;
            }
        }
        return -1;
    }
}

